lib/matplotlib/patches.py

```diff
@@ -1113,6 +1118,11 @@ def __init__(self, x, y, dx, dy, width=0.001, length_includes_head=False,
             head_length = 1.5 * head_width
 
         distance = np.sqrt(dx ** 2 + dy ** 2)
+
+        # Account for divide by zero
+        if distance == 0:
```

Comment:
Yes, that makes sense. Do you want to just move the if statement above the division, or handle it without changing distance? 
```python
if distance != 0:
    cx = float(dx) / distance
    sx = float(dy) / distance
else:
    cx = float(dx)
    sx = float(dy)
```